tests/test_function.py

```diff
@@ -93,17 +93,28 @@ def test_function_list_on_cluster(self, r):
                 [[b"name", b"myfunc", b"description", None, b"flags", [b"no-writes"]]],
             ]
         ]
+        resp3_function_list = [
+            {
+                b"library_name": b"mylib",
+                b"engine": b"LUA",
+                b"functions": [
+                    {b"name": b"myfunc", b"description": None, b"flags": {b"no-writes"}}
+                ],
+            }
+        ]
         primaries = r.get_primaries()
         res = {}
+        resp3_res = {}
         for node in primaries:
             res[node.name] = function_list
-        assert r.function_list() == res
-        assert r.function_list(library="*lib") == res
+            resp3_res[node.name] = resp3_function_list
+        assert_resp_response(r, r.function_list(), res, resp3_res)
+        assert_resp_response(r, r.function_list(library="*lib"), res, resp3_res)
         node = primaries[0].name
-        assert (
-            r.function_list(withcode=True)[node][0][7]
-            == f"#!{engine} name={lib} \n {function}".encode()
-        )
+        code = f"#!{engine} name={lib} \n {function}".encode()
```

Comment:
shouldn't the gears test code not be here? I think this contains part of your discussion with @shacharPash for #2861